src/BPEssentials/Events/OnRegister.cs

```diff
@@ -11,59 +11,19 @@ namespace BPEssentials.RegisteredEvents
+        [Target(GameSourceEvent.ManagerTryRegister, ExecutionMode.Test)]
+        public bool OnTryRegister(SvManager svManager, ConnectionData connectionData)
         {
-            if (ValidateUser(svManager, connectionData))
+            if (Core.Instance.Settings.General.LimitNames && !Regex.IsMatch(connectionData.username, @"^[a-zA-Z0-9_-]+$"))
             {
-                if (svManager.TryGetUserData(connectionData.username, out User playerData))
-                {
-                    if(Core.Instance.Settings.General.DisableAccountOverwrite)
-                    {
-                        svManager.RegisterFail(connectionData.connection, "This Name has already been registerd and this Server has disabled overwriting Accounts!");
-                        return;
-                    }
-                    if (playerData.PasswordHash != connectionData.passwordHash)
-                    {
-                        svManager.RegisterFail(connectionData.connection, "Invalid credentials");
-                        return;
-                    }
-                }
-
-                if (!connectionData.username.ValidCredential())
-                {
-                    svManager.RegisterFail(connectionData.connection, $"Name cannot be registered (min: {Util.minCredential}, max: {Util.maxCredential})");
-                    return;
-                }
-                // -- BPE EXTEND
-                if (Core.Instance.Settings.General.LimitNames && !Regex.IsMatch(connectionData.username, @"^[a-zA-Z0-9_-]+$"))
-                {
-                    svManager.RegisterFail(connectionData.connection, $"Your Username can only contain letters: A-Z a-z 0-9 _ -");
-                    return;
-                }
-                // BPE EXTEND --
-                svManager.AddNewPlayer(connectionData, playerData?.Persistent);
-            }
-        }
-
-        public bool ValidateUser(SvManager svManager, ConnectionData authData)
-        {
-            if (!svManager.HandleWhitelist(authData.username))
-            {
-                svManager.RegisterFail(authData.connection, "Account not whitelisted");
+                svManager.RegisterFail(connectionData.connection, $"Your Username can only contain letters: A-Z a-z 0-9 _ -");
                 return false;
             }
-
-            // Don't allow multi-boxing, WebAPI doesn't prevent this
-            foreach (ShPlayer p in EntityCollections.Humans)
+            if (Core.Instance.Settings.General.DisableAccountOverwrite)
             {
-                if (p.username == authData.username)
-                {
-                    svManager.RegisterFail(authData.connection, "Account still logged in");
-                    return false;
-                }
+                svManager.RegisterFail(connectionData.connection, "This Name has already been registerd and this Server has disabled overwriting Accounts!");

```

Comment:
Minor typo (registerd -> registered)
```suggestion
                svManager.RegisterFail(connectionData.connection, "This name has already been registered and this server has disabled overwriting accounts!");
```